### DISCUSSION TITLE
Migration to Podman ?

### DISCUSSION THREAD

**Discussion by cHJlaXpoZXI** [ThePorgs/Exegol#96] [Q&A]
Hi everybody,

Thanks for this excellent tool ! I have a question regarding Docker : Docker is awesome but have some security issue like the fact that you need root to launch Docker.

Podman is a serious alternative compatible with docker commands. Regarding the docker integration with Python, I have found that Podman have a lib to talk with the API : https://pypi.org/project/podman/ and we can launch podman without root.

What do you think ? If it's not possible, it's OK, maybe try to launch Docker in rootless mode (it's a little bit complex tho...) to mitigate the issue.

Best regards,

**Comment by ShutdownRepo** [comment]
Hey there, thanks for raising this topic. The problem with migrating from docker to podman in this context is that Exegol has been full developed around Docker, either technically or conceptually. Exegol's wrapper's purpose is to manipulate Docker objects. And then we also have Dockerhub's registry and automatic builds.
While I'm not against the idea of migrating from Docker to podman (especially since Docker implies security issues, among other things), in my opinion, it would be so much work that it couldn't be done freely on our spare time like we are doing right now.
What's your opinion on this @Dramelac ?

**Reply by cHJlaXpoZXI** [reply]
> 

Yeah I understand that it cost a lot to modify everything. If you need some help to bring this, I can help a little bit for that too.

Regards !

**Comment by extiop** [comment]
You can manage to use podman with Exegol by pulling your favorite image from the docker hub directly and launching it manually, without the wrapper. Here is how I did it :

The setup :

```bash
#!/bin/bash

podman pull docker.io/nwodtuhs/exegol:ad
podman run --interactive --tty --detach --network host --name exegol docker.io/nwodtuhs/exegol:ad
```

The run (using nushell, the idea is the same) :
```bash
#!/usr/bin/nu

let ctrID = (podman inspect --format={{.Id}} exegol)
podman start $ctrID
podman exec -it $ctrID zsh
```
